### ISSUE TITLE
Broken connectivity while using IPVS, ExternalIP and externalTrafficPolicy: Local

### ISSUE THREAD

**Issue by NottyRu** [kubernetes/kubernetes#124260]
### What happened?

First of all, I read the related issues (https://github.com/kubernetes-sigs/kubespray/issues/10572, https://github.com/k3s-io/k3s/issues/7183, https://github.com/kubernetes/kubernetes/issues/121272, https://github.com/kubernetes/kubernetes/issues/117613) and realized that the problem is still exists in mainstream.

I'm dealing with a Kubernetes cluster managed by Kubespray. It's set up in IPVS mode and uses Calico as a CNI. The cluster is tightly coupled with our network infrastructure: it announces all the types of Kubernetes IPs into ToR switches via BGP (so called IP Fabric Network Design).

In this scheme, the simplest way to assign a globally routed IP into a Service is to declare an ExternalIP. This doesn't require to interact with any kind of Kubernetes API using third-party software (like LoadBalancerIP do), Calico simply catches up the IPs like this and informs the switches about. That's why I can't consider ExternalIPs as an ancient concept (https://github.com/kubernetes/kubernetes/issues/121272#issuecomment-1770986681), they simplify the things.

All the ExternalIPs are automatically assigned to kube-ipvs0 which is great and all work fine with "externalTrafficPolicy: Cluster". But if I change the policy to Local, connections from outside a Kubernetes node to the corresponding ExternalIP become broken with "icmp-port-unreachable" error caused by iptables:

```
-A KUBE-IPVS-FILTER -m set --match-set KUBE-EXTERNAL-IP dst,dst -j RETURN
-A KUBE-IPVS-FILTER -m set --match-set KUBE-HEALTH-CHECK-NODE-PORT dst -j RETURN
-A KUBE-IPVS-FILTER -m conntrack --ctstate NEW -m set --match-set KUBE-IPVS-IPS dst -j REJECT --reject-with icmp-port-unreachable
```

### What did you expect to happen?

It seems the only thing that the externalTrafficPolicy does in the network stack is moving ExternalIPs between KUBE-EXTERNAL-IP and KUBE-EXTERNAL-IP-LOCAL ipsets, but there's no equivalent RETURN rule for the -LOCAL one in the iptables. If I insert the rule manually:

```
-A KUBE-IPVS-FILTER -m set --match-set KUBE-EXTERNAL-IP dst,dst -j RETURN
-A KUBE-IPVS-FILTER -m set --match-set KUBE-EXTERNAL-IP-LOCAL dst,dst -j RETURN
-A KUBE-IPVS-FILTER -m set --match-set KUBE-HEALTH-CHECK-NODE-PORT dst -j RETURN
-A KUBE-IPVS-FILTER -m conntrack --ctstate NEW -m set --match-set KUBE-IPVS-IPS dst -j REJECT --reject-with icmp-port-unreachable
```
"externalTrafficPolicy: Local" works as expected. BTW, I have no idea why we should reject anything here at all, do you?

### How can we reproduce it (as minimally and precisely as possible)?

Deploy a single node with IPVS mode. Declare a Service with ExternalIP and externalTrafficPolicy: Local. See the ExternalIP works locally, but filtered from outside the node.

### Kubernetes version

<details>

```
# kubectl version
WARNING: This version information is deprecated and will be replaced with the output from kubectl version --short.  Use --output=yaml|json to get the full version.
Client Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.5", GitCommit:"93e0d7146fb9c3e9f68aa41b2b4265b2fcdb0a4c", GitTreeState:"clean", BuildDate:"2023-08-24T00:48:26Z", GoVersion:"go1.20.7", Compiler:"gc", Platform:"linux/amd64"}
Kustomize Version: v5.0.1
Server Version: version.Info{Major:"1", Minor:"27", GitVersion:"v1.27.5", GitCommit:"93e0d7146fb9c3e9f68aa41b2b4265b2fcdb0a4c", GitTreeState:"clean", BuildDate:"2023-08-24T00:42:11Z", GoVersion:"go1.20.7", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

Self-managed cluster via Kubespray v2.23.0

### OS version

<details>

```
# cat /etc/os-release 
PRETTY_NAME="Ubuntu 22.04.3 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.3 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

```
# calicoctl version
Client Version:    v3.25.2
Git commit:        978a0e4bc
Cluster Version:   v3.25.2
Cluster Type:      kubespray,bgp,kubeadm,kdd,k8s
```

</details>

**Comment by aroradaman** [comment]
@NottyRu this is already fixed in master https://github.com/kubernetes/kubernetes/commit/19da26005b3d5fedc9cf1370e1d006ac6c939c5c

**Comment by uablrek** [comment]
And cherry-picked to v1.29

**Comment by uablrek** [comment]
This is a duplicate of https://github.com/kubernetes/kubernetes/issues/121909.

It is fixed for v1.30 in https://github.com/kubernetes/kubernetes/pull/121919.

Back-ported to v1.28 and v1.29 in: https://github.com/kubernetes/kubernetes/pull/122323 and https://github.com/kubernetes/kubernetes/pull/121928.

It will not be back-ported to v1.27.

/close
